.github/workflows/binary.yml

```diff
@@ -0,0 +1,290 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [ARM, RISCV64]
+        features: ["", MCS]
+        optimise: ["-O1", "-O2"]
+    steps:
+      # It would be nice if there was a way to prevent the job from starting.
+      - name: Check enabled
+        id: enabled
+        shell: bash
+        env:
+          ARCH: ${{ matrix.arch }}
+          FEATURES: ${{ matrix.features }}
+          OPTIMISE: ${{ matrix.optimise }}
+          ENABLED: ${{ needs.artifacts.outputs.enabled_configs }}
+        run: |
+          # Check whether this configuration is enabled
+          CONFIG="${ARCH}${FEATURES:+-${FEATURES}}"
+          if grep -qx "${CONFIG}" <<< "${ENABLED}"; then
+            echo "C graph-lang found for ${CONFIG}, proceeding with decompilation"
+            echo "::set-output name=config::${CONFIG}"
+            echo "::set-output name=target::${CONFIG}${OPTIMISE}"
+          else
+            echo "No C graph-lang found for ${CONFIG}, skipping decompilation"
+          fi
+      - name: Checkout l4v
+        uses: actions/checkout@v3
+        # We use the ref on which this workflow was triggered,
+        # not the one that caused the trigger.
+        with:
+          path: l4v
+      - name: Checkout Isabelle
+        uses: actions/checkout@v3
+        with:
+          repository: seL4/isabelle
+          ref: ${{ needs.artifacts.outputs.isabelle_commit }}
+          path: isabelle
+      - name: Checkout seL4
+        uses: actions/checkout@v3
+        with:
+          repository: seL4/seL4
+          ref: ${{ needs.artifacts.outputs.sel4_commit }}
+          path: seL4
+      - name: Checkout graph-refine
+        uses: actions/checkout@v3
+        with:
+          repository: seL4/graph-refine
+          ref: ${{ needs.artifacts.outputs.graph_refine_commit }}
+          path: graph-refine
+      - name: Install Nix
+        if: steps.enabled.outputs.config
+        uses: cachix/install-nix-action@v16
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - name: Install Cachix
+        if: steps.enabled.outputs.config
+        uses: cachix/cachix-action@v10
+        with:
+          name: sel4-bv
+          authToken: ${{ secrets.BV_CACHIX_AUTH_TOKEN }}
+      - name: Disable function clones
+        # TODO: Upstream this change to seL4.
+        # This disables some -O2 interprocedural optimisations that
+        # binary verificaation can't handle.
+        working-directory: seL4
+        run: |
+          # Add compile options to gcc.cmake
+          ( echo
+            echo "# Binary verification cannot handle cloned functions."
+            echo "if(KernelVerificationBuild)"
+            echo "    add_compile_options(-fno-partial-inlining -fno-ipa-cp -fno-ipa-sra)"
+            echo "endif()"
+          ) >> gcc.cmake
+      - name: Build target
+        if: steps.enabled.outputs.config
+        working-directory: graph-refine
+        shell: nix-shell --run "bash -eo pipefail {0}"
```

Comment:
I guess this is where `nix` comes in, but it's still unclear to me what exactly is going on there.

Reply:
This is running the `run` script under a sub-shell which provides the tools needed by the `run` script. `nix-shell` sets up the environment according to the [`shell.nix`](https://github.com/seL4/graph-refine/blob/ci-riscv64/shell.nix) in the working directory.